### PR TITLE
Use the new screensaver class in Pam's, Logic, and Euclid scripts

### PR DESCRIPTION
**Pull request by chrisib**
These changes were originally part of https://github.com/Allen-Synthesis/EuroPi/pull/309 but were broken out to simplify that original MR.

This MR depends on the updated screensaver implemented in 309. Once 309 is approved and merged into `main` I'll rebase to reduce the number of commits in this branch.

### CODE REVIEW

**Comment by chrisib** [comment]
Rebased to latest `main` after merging #309

File: software/contrib/kompari.py
Comment (roryjamesallen) [inline software/contrib/kompari.py:1]: This doesn't seem to use the new screensaver class

File: software/contrib/pams.py [hunk 1]
```diff
@@ -1397,23 +1385,17 @@ def main(self):
...
 
-            elapsed_time = time.ticks_diff(now, self.last_interaction_time)
-            if elapsed_time > BLANK_TIMEOUT_MS:
-                self.screensaver.draw_blank()
-            elif elapsed_time > SCREENSAVER_TIMEOUT_MS:
-                self.screensaver.draw()
-            else:
-                oled.fill(0)
-                self.main_menu.draw()
+            ssoled.fill(0)
```
Comment (roryjamesallen) [inline software/contrib/pams.py:1388]: might as well use .clear() to show off the extrra function of ssoled!

**Review by roryjamesallen — CHANGES_REQUESTED** [review]
The only change request is to kompari, either to implement the screensaver or just to remove the commits and not include it in this PR as it barely uses the oled

File: software/contrib/pams.py (hunk 1 above)
Comment (chrisib) [reply software/contrib/pams.py:1388]: `.clear()` was removed; it shouldn't have existed in the first place.

File: software/contrib/kompari.py
Comment (chrisib) [reply software/contrib/kompari.py:1]: I think that was another victim of rebasing across branches.

In any case, I've updated Kompari to use the screensaver now too.

**Review by mjaskula — APPROVED** [review]
These changes look good to me.

**Pull request merged**